Set repeater rate limit delay in settings.py

gherceg opened the pull request:
## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->

## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
Would be easier to tune these values if they lived in settings instead of being hard coded.

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.

In particular consider how existing data may be impacted by this change.
-->

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->


### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

gherceg committed:
Set repeater rate limit delay in settings.py

mjriley commented on corehq/motech/repeaters/const.py:
```diff
@@ -6,7 +6,10 @@
 MAX_RETRY_WAIT = timedelta(days=7)
 MIN_RETRY_WAIT = timedelta(minutes=60)
-RATE_LIMITER_DELAY_RANGE = (timedelta(minutes=0), timedelta(minutes=15))
+RATE_LIMITER_DELAY_RANGE = (
+    timedelta(minutes=settings.MIN_REPEATER_RATE_LIMIT_DELAY),
+    timedelta(minutes=settings.MAX_REPEATER_RATE_LIMIT_DELAY),
+)
```
I'm curious how you feel about defaults? When we reference these values directly, they need to be present in our `settings` object or an error is thrown. I've been looking at using syntax like `getattr(settings, 'MAX_REPEATER_RATE_LIMIT_DELAY', 15)` instead, as it avoids the need to put everything in settings.  At a certain point, when we place everything in settings, I'd imagine the settings file becomes pretty overwhelming. Provided we have sensible defaults, using `getattr` reduces the overall settings size at the cost of some discoverability.

I wonder if the divide is between settings the user will know they need to tweak without any knowledge of our system (i.e. custom URLs) vs technical details. Not sure where this should fall.

gherceg replied:
Yeah I like the sound of that better. I would think this is a technical detail that users will likely never need to think about so going with your approach seems objectively better.

gherceg replied:
> the cost of some discoverability.

I suppose if we were really concerned about this, we could create some helper function to do exactly this, but just makes it easy to grep for. Though for now, I'm not too concerned about that.

gherceg replied:
52d49d8cf0a6e516b531a7e696cf9a28dd98e62f

mjriley reviewed (approved):
Approving -- left a non-blocking comment

gherceg committed:
Use defaults with getattr rather than in settings

This avoids polluting settings.py with default values that a user will
likely never actually need to tweak.
